File: Document/beginners/sample_mruby/sample00/main_task.rb
class MainTask

  PWM_MAX = 100
  PWM = PWM_MAX / 6
  DURATION = 2000

  def initialize
    @left_wheel = EV3RT_TECS::Motor.new(:port_c)
    @right_wheel = EV3RT_TECS::Motor.new(:port_b)
    # C++の__FILE__の代わりにクラス名を表示する
    Util::Display.init("sample00 " + self.class.to_s)
  end

  def execute
   loop do
      Util::Display.message("Forwarding...", 2)
      @left_wheel.power = PWM
      @right_wheel.power = PWM
      EV3RT_TECS::RTOS.delay(DURATION)

      Util::Display.message("Backwarding...", 2)
      @left_wheel.power = -PWM
      @right_wheel.power = -PWM
      EV3RT_TECS::RTOS.delay(DURATION)

      # 左ボタンを長押し、それを捕捉する
      break if EV3RT_TECS::Button[:left].pressed?      
    end

    Util::Display.message("Stopped.", 2)
    @left_wheel.stop
    @right_wheel.stop
    # 左ボタンの長押しが終わるまで待つ
    while EV3RT_TECS::Button[:left].pressed?
    end
  end

end

MainTask.new.execute

File: Document/beginners/sample_java/sample00/src/jp/etrobo/ev3/sample00/MainTask.java
/*
 *  MainTask.java (for leJOS EV3)
 *  Created on: 2016/06/11
 *  Copyright (c) 2016 Embedded Technology Software Design Robot Contest
 */
package jp.etrobo.ev3.sample00;

import lejos.hardware.Button;
import lejos.hardware.lcd.LCD;
import lejos.hardware.port.MotorPort;
import lejos.hardware.port.TachoMotorPort;
import lejos.utility.Delay;

/**
 * leJOS EV3 用 Java
 */
public class MainTask {

    /**
     * メイン
     */
    public static void main(String[] args) {
        TachoMotorPort leftWheel = MotorPort.C.open(TachoMotorPort.class);
        TachoMotorPort rightWheel = MotorPort.B.open(TachoMotorPort.class);

    	int pwm = TachoMotorPort.MAX_POWER / 6;
    	long duration = 2000;

    	LCD.drawString("MainTask", 0, 2);
        while(true){

        	LCD.drawString("Forwarding...", 0, 4);
        	leftWheel.controlMotor(pwm, TachoMotorPort.PWM_FLOAT);
        	rightWheel.controlMotor(pwm, TachoMotorPort.PWM_FLOAT);
        	Delay.msDelay(duration);

        	LCD.drawString("Backwarding...", 0, 4);
        	leftWheel.controlMotor(-pwm, TachoMotorPort.PWM_FLOAT);
        	rightWheel.controlMotor(-pwm, TachoMotorPort.PWM_FLOAT);
        	Delay.msDelay(duration);

            if (Button.LEFT.isDown()) {
            	break;
            }
        }

        LCD.drawString("Stopped.", 0, 4);
        leftWheel.setPWMMode(TachoMotorPort.STOP);
        rightWheel.setPWMMode(TachoMotorPort.STOP);
        while (Button.LEFT.isDown()) {
        	;
        }
    }
}